pkg/cloudprovider/providers/aws/aws.go

```diff
@@ -3778,18 +3778,16 @@ func (c *Cloud) updateInstanceSecurityGroupsForLoadBalancer(lb *elb.LoadBalancer
 	}
 
 	// Determine the load balancer security group id
-	loadBalancerSecurityGroupID := ""
-	for _, securityGroup := range lb.SecurityGroups {
-		if aws.StringValue(securityGroup) == "" {
-			continue
-		}
-		if loadBalancerSecurityGroupID != "" {
-			// We create LBs with one SG
-			glog.Warningf("Multiple security groups for load balancer: %q", aws.StringValue(lb.LoadBalancerName))
-		}
-		loadBalancerSecurityGroupID = *securityGroup
+	request := &ec2.DescribeSecurityGroupsInput{
+		GroupNames: []*string{lb.SourceSecurityGroup.GroupName},
+	}
+	sg, err := c.ec2.DescribeSecurityGroups(request)
+	if err != nil {
+		return fmt.Errorf("error querying security groups: %q", err)
 	}
-	if loadBalancerSecurityGroupID == "" {
+	// err is nil, we have at least 1 security group
+	loadBalancerSecurityGroupID := aws.StringValue(sg[0].GroupId)
```

Comment:
It's a slightly dangerous pattern, IMO.  Simpler to program more defensively, I think, with an explicit `if len(sg) == 0` check